internal/observability/call.go

```diff
@@ -234,6 +234,13 @@ func (c call) endStats(
 	applicationErrorMeta *transport.ApplicationErrorMeta,
 ) {
 	c.edge.calls.Inc()
+
+	if c.direction == _directionInbound {
+		if deadlineTime, ok := c.ctx.Deadline(); ok {
+			c.edge.ttls.Observe(deadlineTime.Sub(c.started))
+		}
+	}
+
```

Comment:
This will not emit the TTL metrics explained in this ticket: https://code.uberinternal.com/T4884039 
Here this metric will emit the remaining time the request would have. While based on the ticket description we need the following metrics:

```
yarpc.edge.ttl	Histogram of timeouts passed in by the caller for all calls.
yarpc.edge.timeout-ttl	Histogram of timeouts passed in by the caller for calls that failed with a timeout.
```

Reply:
What I said above is wrong, this is good for `yarpc.edge.ttl`!

We are missing yarpc.edge.timeout-ttl though

Reply:
I have added changes for `timeout-ttl`